jetty-core/jetty-websocket/jetty-websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/annotations/OnWebSocketMessage.java

```diff
@@ -23,42 +23,38 @@
 
 /**
  * <p>Annotation for methods to receive BINARY or TEXT WebSocket events.</p>
- * <p>Acceptable method patterns:</p>
- * <u>Text Message Versions</u>
+ * <p>Acceptable method signatures for text messages:</p>
  * <ol>
- * <li>{@code public void methodName(String text)}</li>
- * <li>{@code public void methodName(Session session, String text)}</li>
- * <li>{@code public void methodName(Reader reader)}</li>
- * <li>{@code public void methodName(Session session, Reader reader)}</li>
+ * <li>{@code public void methodName(Session session, *String text)}</li>
+ * <li>{@code public void methodName(Session session, *Reader reader)}</li>
  * </ol>
+ * <p>The {@code *} before the parameter type means that the parameter is mandatory.</p>
  * <p>NOTE</p>
  * <p>Method that takes a {@link Reader} must have
  * {@link WebSocket#autoDemand()} set to {@code true}.</p>
```

Comment:
Not related to your changes but I think this is wrong. 

You can have `WebSocket#autoDemand() = false`, and then you are responsible for demanding the next message once `onWebSocketMessage` is called.

The `MessageInputStream` will demand until the end of each message regardless of the `autoDemand` setting.